can-define.js

```diff
@@ -859,7 +859,13 @@ define.types = {
 	 * any string, including "", is truthy.
 	 */
 	'htmlbool': function(val) {
-		return typeof val === "string" || !!val;
+		if (val === '') {
+			return true;
+		}
+		if (val == null) {
```

Comment:
ensure no participation?  Do you mean comply with 2316?  I thought this was an exception.

This is for scenarios like:

```js
VM = DefineMap.extend({
  prop: "htmlbool"
})
```

```
<component {{#something}}prop{{/something}}>
```

In this case, `prop` will be set to `""` so need to be `true`.

I'm not sure what the following should do:

```
<component {{#something}}prop="false"{{/something}}>
```

perhaps setting to false is right.

What you might want to use is https://canjs.com/doc/can-util/js/string-to-any/string-to-any.html



```
!!stringToAny(value)
```